app/src/main/java/net/gsantner/opoc/util/FileUtils.java

```diff
@@ -60,6 +61,37 @@ public static String readTextFileFast(final File file) {
         return "";
     }
 
+    public static Pair<String, Boolean> readTextFileFastWithBOM(final File file) {
```

Comment:
preferably we don't stack read methods onTop onTop onTop with different features.   
We already had that and the current focus on simplifying and debloating.

So what I rather suggest is, have one such function, and have a Pair<String, Map> as result. And in that Map any metadata can be put. With the first case here being BOM information.

What do you think.

Reply:
I'm not familiar with Java. Thanks for your advice.

Reply:
Ah, I see why we chose the map approach :)